rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -785,137 +827,181 @@ def test_missing_acquisitions(accountant):
     assert cost_basis.missing_acquisitions == expected_missing_acquisitions
 
 
-def test_accounting_average_cost_basis(accountant):
+def test_accounting_average_cost_basis_simple(accountant):
     """
     Test data is gotten from:
     https://www.adjustedcostbase.ca/blog/how-to-calculate-adjusted-cost-base-acb-and-capital-gains/
     """
-    asset = A_ETH
-    cost_basis = accountant.pots[0].cost_basis
+    pot = accountant.pots[0]
+    events = pot.processed_events
+    cost_basis = pot.cost_basis
     cost_basis.reset(DBSettings(cost_basis_method=CostBasisMethod.ACB))
-    asset_events = cost_basis.get_events(asset)
+    asset_events = cost_basis.get_events(A_ETH)
+    manager = asset_events.acquisitions_manager
+    expected_average_asset_value = ZERO
+    expected_remaining_amount = ZERO
 
     # check that the average cost basis is calculated properly
     # whenever a spend happens after an acquisition
-    assert asset_events.acquisitions_manager.remaining_amount == ZERO
-    event1 = AssetAcquisitionEvent(
-        amount=FVal(100),
-        timestamp=1,
-        rate=FVal(50),
-        index=1,
-    )
-    asset_events.acquisitions_manager.add_acquisition(event1)
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(100)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(5000)
-    cost_basis_result = asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(50),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    )
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(50)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(2500)
-    assert cost_basis_result.taxable_bought_cost == FVal(2500)
-    assert cost_basis_result.is_complete is True
+    add_acquisition(pot, FVal(100), price=FVal(50))
+    expected_average_asset_value = events[0].price
+    expected_remaining_amount = events[0].taxable_amount
+    assert_acb_state(manager, expected_average_asset_value, expected_remaining_amount)
+    add_spend(pot, FVal(50))
+    last_cost_basis = events[-1].cost_basis
+    assert last_cost_basis.taxable_bought_cost == expected_average_asset_value * last_cost_basis.taxable_amount  # noqa: E501
```

Comment:
1. Adjust docstring
2. You are focusing on the totally wrong thing when testing. you seem to be lost in the (simple) math of the ACB instead of focusing on the one thing that matters. **what is the profit/loss** for the user at each sell.
3. Please write proper comments to make it easier to understand as you go and see what you should expect:
```
Buy 100 ETH for $50  acb: $50
Sell 50 ETH for $1  PNL: 50 - 50*50 = -2450
```

And use proper scenarios. This is crazy as a loss. And if you check the result of the PnL for the event in your test it will show `-2500` which is wrong. Use simple stuff like, buy something and then sell at a profit, then sell at a loss.

Reply:
Changed the tests. Hope they are better now.